### ISSUE TITLE
Improve back-office custom field inputs accessibility and ergonomy

### ISSUE THREAD

**Issue by charleshopfner** [vendure-ecommerce/vendure#2703]
**Is your feature request related to a problem? Please describe.**
The inputs in the Vendure Back-office lack certain accessibility features and could be improved in terms of behavior and usability.

**Describe the solution you'd like**
I would like to enhance the custom fields inputs in the Vendure Back-office to improve accessibility, behavior, and usability. This would involve adding features such as the ability to include captions and labels for the fields, add visual distinction for the differents input states (readonly, ...) , and enhancing overall user experience.

**Describe alternatives you've considered**
One alternative solution could be to rely on third-party libraries to achieve similar functionality. However, integrating these may introduce dependencies and increase maintenance overhead.

**Additional context**
Depending on the ids of the custom fields, the displayed name isn't always easily understandable for all the back-office users.
Also, there is no visual disctinction made between readonly fields and editable fields. As example, the "is" field below is readonly, and the "asics" field is editable :
![image](https://github.com/vendure-ecommerce/vendure/assets/99817995/34a42709-fe12-419b-96df-bb3b069c9ef5)

**Comment by michaelbromley** [comment]
Hi!

Let's split this into the concrete changes:

1. (fix) Make readonly state easily visually discernable
2. (feat) Ability to add caption to a custom field. Can you give an example use-case for this?
3. (feat) Ability to add a label to a custom field. Is this different from what already exists with the [label property](https://docs.vendure.io/guides/developer-guide/custom-fields/#label)?

> One alternative solution could be to rely on third-party libraries to achieve similar functionality. However, integrating these may introduce dependencies and increase maintenance overhead.

100% agree on this.

**Comment by charleshopfner** [comment]
Hi,
Thanks for your answer. 
For the 2) : as example, we encountered a custom field for the current electricity price and we used a text input for this. It would have been interesting to add a helper below the field to indicate what price format is needed.
More generally, it is also important for accessibility to help the user to fill the forms.

For the 3): My bad, i forgot this property, and yeah this is what I was talking about... We didn't use it because we couldn't manage the translation in a .json file and opened [this issue](https://github.com/orgs/vendure-ecommerce/projects/1/views/1?pane=issue&itemId=39037198) at the time.

Another thing is coming to my mind :
Adding a symbol at the end of the label for the required fields.
At the moment, there is no distinction between required and non-required fields.

Thanks for the infos ! If it is ok for you, i'll start working on this

**Comment by dlhck** [comment]
Could be done together with #2903 in case @charleshopfner is not providing a PR for that.

**Comment by dlhck** [comment]
This feature request has been moved to GitHub Discussions for community voting.
Please upvote the discussion if you would like to see this implemented: https://github.com/vendurehq/vendure/discussions/4265
